src/Parser/VariadicMethodsVisitor.php

```diff
@@ -0,0 +1,133 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Parser;
+
+use PhpParser\Node;
+use PhpParser\Node\Name;
+use PhpParser\Node\Stmt\ClassMethod;
+use PhpParser\NodeVisitorAbstract;
+use PHPStan\Reflection\ParametersAcceptor;
+use PHPStan\TrinaryLogic;
+use function array_key_exists;
+use function array_pop;
+use function implode;
+use function in_array;
+
+final class VariadicMethodsVisitor extends NodeVisitorAbstract
+{
+
+	private ?Node $topNode = null;
+
+	private ?string $inNamespace = null;
+
+	private ?string $inClassLike = null;
+
+	/** @var array<string> */
+	private array $classStack = [];
+
+	private ?string $inMethod = null;
+
+	/** @var array<string, array<string, TrinaryLogic>> */
+	private array $variadicMethods = [];
+
+	public const ATTRIBUTE_NAME = 'variadicMethods';
+
+	public function beforeTraverse(array $nodes): ?array
+	{
+		$this->topNode = null;
+		$this->variadicMethods = [];
+		$this->inNamespace = null;
+		$this->classStack = [];
+		$this->inClassLike = null;
+		$this->inMethod = null;
+
+		return null;
+	}
+
+	public function enterNode(Node $node): ?Node
+	{
+		if ($this->topNode === null) {
+			$this->topNode = $node;
+		}
+
+		if ($node instanceof Node\Stmt\Namespace_ && $node->name !== null) {
+			$this->inNamespace = $node->name->toString();
+		}
+
+		if (
+			$node instanceof Node\Stmt\Class_
+			|| $node instanceof Node\Stmt\ClassLike
+		) {
+			if (!$node->name instanceof Node\Identifier) {
+				$className = 'class@anonymous:' . $node->getStartLine();
```

Comment:
This isn't sufficiently unique. This was recently fixed elsewhere by https://github.com/phpstan/phpstan-src/pull/3328.